instance/tests/models/factories/openedx_instance.py

```diff
@@ -37,5 +39,14 @@ class OpenEdXInstanceFactory(DjangoModelFactory):
     class Meta:
         model = OpenEdXInstance
 
-    sub_domain = factory.Sequence('instance{}.test'.format)
+    @classmethod
+    def create(cls, *args, **kwargs):
+        # OpenEdXInstance constructor accepts either a 'sub_domain' or 'instance_lms_domain' value. Only generate a
+        # random value for 'internal_lms_domain' if neither 'sub_domain' nor 'internal_lms_domain' are provided.
+        if 'sub_domain' not in kwargs and 'internal_lms_domain' not in kwargs:
+            kwargs = kwargs.copy()
+            domain = generate_internal_lms_domain('instance{}.test'.format(str(uuid.uuid4())[:8]))
```

Comment:
@mtyaka Same comment here about splitting this up into two/three lines for readability.